core/pom.xml

```diff
@@ -242,6 +242,13 @@
       </plugins>
     </pluginManagement>
     <plugins>
+      <plugin>
+        <groupId>com.github.spotbugs</groupId>
+        <artifactId>spotbugs-maven-plugin</artifactId>
+        <configuration>
+          <omitVisitors>SharedVariableAtomicityDetector</omitVisitors>
```

Comment:
We have other spotbugs filters in src/main/spotbugs/exclude-filter.xml, where we have fine-grained control over things to filter, and instead of omitting entire detectors, we can filter based on the documented bug names. I'm not sure which is better, but it would probably be nice to consolidate the config, so we're doing it all in one place, instead of doing some here in the pom, and others elsewhere in the exclude-filter.xml file.

Reply:
It's only temporary while it's in the PR. Once it gets merged, it's more or less permanent, or at least as permanent as the exclude-filter.xml files. My review comments are based on the desired end result after this is merged. That is to say, doing it in a different way while you're working on it is fine... but unless there's a good reason to have these configured separately from what's in the exclude-filter.xml files, then it's better that the end result by the time this is merged, has removed these extra configs, or consolidated them with the exclude-filter.xml files, so we don't have config scattered everywhere that basically does similar things.